app/src/main/java/jp/co/cyberagent/stf/Service.java

```diff
@@ -367,4 +373,73 @@ public void run() {
             }
         }
     }
+
+    /**
+     * Monitors the adb state by checking /sys/class/android_usb/android0/state
+     * <p>
+     * If state is DISCONNECTED then IdentityActivity will be shown to allow easier identification
+     * of malfunctioning devices
+     */
+    private class AdbMonitor extends Thread {
+        // If something goes wrong you have 30 seconds to kill the STFService
+        // Using smaller numbers will basically lock the devices until usb connection is established
+        private static final int INTERVAL_MS = 30000;
+
+        @Override
+        public void run() {
+            Log.d(TAG, "Starting adb monitor thread");
+            java.lang.Process process;
+            String state = "";
+            try {
+                while (!isInterrupted()) {
+                    /**
+                     * In order for the monitor to work you need to grant DUMP permission for
+                     * the apk, e.g.
+                     *
+                     * adb -d shell pm grant jp.co.cyberagent.stf android.permission.DUMP
+                     */
+                    int dumpPermission = ContextCompat.checkSelfPermission(getApplication(), Manifest.permission.DUMP);
+                    if (dumpPermission == PackageManager.PERMISSION_GRANTED) {
+                        String[] cmd = {
+                            "/system/bin/dumpsys",
+                            "usb"
+                        };
+
+                        process = Runtime.getRuntime().exec(cmd);
+
+                        /**
+                         * If the output of the command will change then by default device will be
+                         * considered connected
+                         */
+                        String currentState = "";
+                        BufferedReader adbdStateReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+                        for (String line = adbdStateReader.readLine(); line != null; line = adbdStateReader.readLine()) {
+                            if (line.contains("Kernel state:")) {
+                                currentState = line.split(":").length == 2 ? line.split(":")[1] : "";
+                                break;
+                            }
+                        }
+
+                        if (!currentState.equals(state)) {
+                            Log.d(TAG, "Kernel state changed to" + currentState);
+                            state = currentState;
+                        }
+                        boolean disconnected = state.contains("DISCONNECTED");
```

Comment:
Super minor thing but the formatting is throwing me off here, could you add a blank line before the variable and remove the one after it so that it matches the rest of the code.

Reply:
Done